src/classlibnative/bcltype/system.cpp

```diff
@@ -355,18 +355,6 @@ INT32 QCALLTYPE SystemNative::GetProcessorCount()
     return processorCount;
 }
 
-FCIMPL0(FC_BOOL_RET, SystemNative::HasShutdownStarted)
-{
-    FCALL_CONTRACT;
-
-    // Return true if the EE has started to shutdown and is now going to
-    // aggressively finalize objects referred to by static variables OR
-    // if someone is unloading the current AppDomain AND we have started
-    // finalizing objects referred to by static variables.
-    FC_RETURN_BOOL(g_fEEShutDown & ShutDown_Finalize2);
```

Comment:
Why shouldn't Environment.HasShutdownStarted continue to return g_fEEShutDown?

Reply:
I'd like to hear this one too.

Reply:
With the finalization on shutdown removed, this can return true only for a very brief moment before the process gets killed. I do not think you can do anything useful with property like that, and all uses of it will be prone to races anyway.